data-prepper-plugins/key-value-processor/src/test/java/com/amazon/dataprepper/plugins/processor/keyvalue/KeyValueProcessorTests.java

```diff
@@ -0,0 +1,237 @@
+
+        assertThat(editedRecords.size(), equalTo(1));
+        assertThat(editedRecords.get(0), notNullValue());
+        assertThat(editedRecords.get(0).getData().toJsonString(),
+                equalTo("{\"message\":\"key1=value1   &key2=value2\",\"parsed_message\":{\"key1\":\"value1\",\"key2\":\"value2\"}}"));
+    }
+
+    @Test
+    void testTrimValueAndKeyRegexKvProcessor() {
+        //override setup
+        keyValueProcessor = new KeyValueProcessor(pluginMetrics, getConfig("", null, "&", "=", "\\s", "\\s"));
+
+        final Map<String, Object> testData = new HashMap();
+        testData.put("message", "key1  =value1   & key2 = value2 ");
+        final Record<Event> record = buildRecordWithEvent(testData);
+
+        final List<Record<Event>> editedRecords = (List<Record<Event>>)keyValueProcessor.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.size(), equalTo(1));
+        assertThat(editedRecords.get(0), notNullValue());
+        assertThat(editedRecords.get(0).getData().toJsonString(),
+                equalTo("{\"message\":\"key1  =value1   & key2 = value2 \",\"parsed_message\":{\"key1\":\"value1\",\"key2\":\"value2\"}}"));
+    }
+
+    @Test
+    void testShutdownIsReady() {
+        assertThat(keyValueProcessor.isReadyForShutdown(), is(true));
+    }
+
+    private void reflectivelySetField(final KeyValueProcessorConfig keyValueProcessorConfig, final String fieldName, final Object value) throws NoSuchFieldException, IllegalAccessException {
+        final Field field = KeyValueProcessorConfig.class.getDeclaredField(fieldName);
+        try {
+            field.setAccessible(true);
+            field.set(keyValueProcessorConfig, value);
+        } finally {
+            field.setAccessible(false);
+        }
+    }
+
+    private KeyValueProcessorConfig getConfig(final String prefix, final String nonMatchValue,
+                                              final String fieldDelimiterRegex, final String keyValueDelimiterRegex, final String trimKeyRegex,
+                                              final String trimValueRegex) {
+        KeyValueProcessorConfig config = new KeyValueProcessorConfig();
+
+        try {
+            reflectivelySetField(config, "prefix", prefix);
+            reflectivelySetField(config, "nonMatchValue", nonMatchValue);
+            reflectivelySetField(config, "fieldDelimiterRegex", fieldDelimiterRegex);
+            reflectivelySetField(config, "keyValueDelimiterRegex", keyValueDelimiterRegex);
+            reflectivelySetField(config, "trimKeyRegex", trimKeyRegex);
+            reflectivelySetField(config, "trimValueRegex", trimValueRegex);
+        } catch (NoSuchFieldException e) {
+            e.printStackTrace();
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+
+        return config;
+    }
+
+    static Record<Event> buildRecordWithEvent(final Map<String, Object> data) {
```

Comment:
You could improve these tests by using a mock here. All of your tests have to assert JSON string values. This can be brittle depending on how it serializes.

If you used Mockito's `verify` functionality, you can ensure that the correct values are _set_ on the Event, rather than checking that the Event itself is complete.